src/EditorFeatures/Core.Wpf/StringIndentation/StringIndentationAdornmentManager.cs

```diff
@@ -45,20 +45,16 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
                 var tagSpans = TagAggregator.GetTags(changedSpan);
                 foreach (var tagMappingSpan in tagSpans)
                 {
-                    if (!ShouldDrawTag(changedSpan, tagMappingSpan, out _))
-                        continue;
-
-                    if (!TryMapToSingleSnapshotSpan(tagMappingSpan.Span, TextView.TextSnapshot, out var span))
-                        continue;
-
-                    if (!TryMapHoleSpans(tagMappingSpan.Tag.OrderedHoleSpans, out var orderedHoleSpans))
-                        continue;
-
-                    if (VisibleBlock.CreateVisibleBlock(span, orderedHoleSpans, TextView) is not VisibleBlock block)
+                    if (!TryGetMappedPoint(changedSpan, tagMappingSpan, out _) ||
+                        !ShouldDrawTag(tagMappingSpan) ||
```

Comment:
the major change here is that we dont' expect to know the text line of the end of the tag (like the other adornments) It's fine if that's offscreen.